code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -426,68 +411,70 @@
 						internal_output.assert_gases(/datum/gas/antinoblium)
-						internal_output.gases[/datum/gas/antinoblium][MOLES] += 0.1 * (scaled_helium / (fuel_injection_rate * 0.0065)) * delta_time
+						internal_output.gases[/datum/gas/antinoblium][MOLES] += 0.9 * (scaled_fuel_list[scaled_fuel_list[3]] / (fuel_injection_rate * 0.0065)) * delta_time
 				if(6)
 					var/scaled_production = clamp(heat_output * 1e-7, 0, fuel_consumption_rate) * delta_time
-					if(m_plasma > 30)
+					moderator_internal.gases[selected_fuel.secondary_products[5]][MOLES] += scaled_production * 0.35
+					moderator_internal.gases[selected_fuel.secondary_products[6]][MOLES] += scaled_production
+					if(moderator_list[/datum/gas/plasma] > 30)
 						moderator_internal.gases[/datum/gas/bz][MOLES] += scaled_production * 1.15
 						moderator_internal.gases[/datum/gas/plasma][MOLES] -= min(moderator_internal.gases[/datum/gas/plasma][MOLES], scaled_production * 1.45)
-					if(m_proto_nitrate)
+					if(moderator_list[/datum/gas/proto_nitrate])
 						internal_output.assert_gases(/datum/gas/zauker, /datum/gas/stimulum)
 						internal_output.gases[/datum/gas/zauker][MOLES] += scaled_production * 5.35
 						internal_output.gases[/datum/gas/stimulum][MOLES] += scaled_production * 2.15
 						moderator_internal.gases[/datum/gas/proto_nitrate][MOLES] -= min(moderator_internal.gases[/datum/gas/proto_nitrate][MOLES], scaled_production * 3.35)
 						radiation *= 2
 						heat_output *= 2.25
-					if(m_bz)
+					if(moderator_list[/datum/gas/bz])
 						for(var/mob/living/carbon/human/human in view(src, HALLUCINATION_HFR(heat_output)))
 							//mesons won't protect you at fusion level 6
 							var/distance_root = sqrt(1 / max(1, get_dist(human, src)))
 							human.hallucination += power_level * 150 * distance_root
 							human.hallucination = clamp(human.hallucination, 0, 200)
-						moderator_internal.gases[/datum/gas/antinoblium][MOLES] += clamp((scaled_helium / (fuel_injection_rate * 0.0045)), 0, 10) * delta_time
-					if(m_healium > 100)
+						moderator_internal.gases[/datum/gas/antinoblium][MOLES] += clamp((scaled_fuel_list[scaled_fuel_list[3]] / (fuel_injection_rate * 0.0045)), 0, 10) * delta_time
+					if(moderator_list[/datum/gas/healium] > 100)
 						if(critical_threshold_proximity > 400)
-							critical_threshold_proximity = max(critical_threshold_proximity - (m_healium / 100 * delta_time ), 0)
+							critical_threshold_proximity = max(critical_threshold_proximity - (moderator_list[/datum/gas/healium] / 100 * delta_time ), 0)
 							moderator_internal.gases[/datum/gas/healium][MOLES] -= min(moderator_internal.gases[/datum/gas/healium][MOLES], scaled_production * 20)
-					internal_fusion.gases[/datum/gas/antinoblium][MOLES] += 0.01 * (scaled_helium / (fuel_injection_rate * 0.0095)) * delta_time
+					internal_fusion.gases[/datum/gas/antinoblium][MOLES] += 0.01 * (scaled_fuel_list[scaled_fuel_list[3]] / (fuel_injection_rate * 0.0095)) * delta_time
```

Comment:
Why is the hardcoded 3 needed here?

Reply:
is the first primary product that you put in, so if you use trit and h2 it will be helium, if you use plasma and o2 it will be co2, that way antinoblium production can be tied to those gases